src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/StaticWebAssets/StaticWebAssetsFileProviderTests.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Tests.StaticWebAssets
+{
+    public class StaticWebAssetsFileProviderTests
+    {
+        [Fact]
+        public void StaticWebAssetsFileProvider_ConstructorThrows_WhenPathIsNotFound()
+        {
+            // Arrange, Act & Assert
+            var provider = Assert.Throws<DirectoryNotFoundException>(() => new StaticWebAssetsFileProvider("/prefix", "/nonexisting"));
```

Comment:
What does this error message look like? Is this the default exception thrown by .NET? Should we try to provide some additional context about why the user is seeing this error?

Reply:
No. The original exception contains all the information and we would simply obfuscate things if we provided our own.